packages/compass-serverstats/src/components/detailview-component.jsx

```diff
@@ -1,165 +0,0 @@
-const React = require('react');
-const Actions = require('../actions');
-const { Button, Icon } = require('@mongodb-js/compass-components');
-const PropTypes = require('prop-types');
-const { withTelemetry } = require('@mongodb-js/compass-telemetry/provider');
-
-class DetailViewComponent extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { error: null, data: {}, display: 'none' };
-  }
-
-  componentDidMount() {
-    this.unsubscribeShowOperationDetails = Actions.showOperationDetails.listen(
-      this.show.bind(this)
-    );
-    this.unsubscribeHideOperationDetails = Actions.hideOperationDetails.listen(
-      this.hide.bind(this)
-    );
-  }
-
-  componentWillUnmount() {
-    this.unsubscribeShowOperationDetails();
-    this.unsubscribeHideOperationDetails();
-  }
-
-  /**
-   * Set the component to visible.
-   *
-   * @param {Object} data - The operation data.
-   */
-  show(data) {
-    this.setState({ data: data, display: 'block' });
-  }
-
-  /**
-   * Set the component to hidden.
-   */
-  hide() {
-    this.setState({ data: {}, display: 'none' });
-  }
-
-  killOp() {
-    this.props.logger.track('DetailView killOp');
-    Actions.killOp(this.state.data.opid);
-    this.hideOperationDetails();
-  }
-
-  /**
-   * Fire the show operation detail action with the row data.
-   */
-  hideOperationDetails() {
-    this.props.logger.track('DetailView hideOperationDetails');
-    Actions.hideOperationDetails();
-  }
-
-  removeMs(key, value) {
-    if (key === 'ms_running') {
-      return undefined;
-    }
-    return value;
-  }
-
-  renderError() {
-    return (
-      <div style={{ display: this.state.display }}>
-        this.state.error.message
-      </div>
-    );
-  }
-
-  renderZero() {
-    return <div style={{ display: this.state.display }} />;
-  }
```

Comment:
I don't see these two views in the refactored component. Is the error state not needed here?

Reply:
Ahhh, sorry, missed this in the PR description, I can see now it's not used